landing-strip/core/src/main/java/com/novoda/landingstrip/TabsState.java

```diff
@@ -0,0 +1,189 @@
+package com.novoda.landingstrip;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.novoda.landingstrip.LandingStrip.TabSetterUpper;
+
+public class TabsState implements OnPagerAdapterChangedListener {
```

Comment:
Not sure how I feel about the name of this, for me state is `getters` and `setters` with no behaviour, can't think of a name though!

maybe because it is doing toooo many things? If only the `attach` was public it would be clearer